geomstats/geometry/general_linear.py

```diff
@@ -200,17 +200,14 @@ class SquareMatrices(MatrixLieAlgebra):
 
     def __init__(self, n, equip=True):
         self.n = n
-        super().__init__(dim=n**2, representation_dim=n, equip=equip)
-        self._mat_space = Matrices(n, n, equip=False)
+        super().__init__(n=n, m=n, equip=equip)
 
-    @staticmethod
-    def default_metric():
-        """Metric to equip the space with if equip is True."""
-        return MatricesMetric
 
-    def _create_basis(self):
-        """Create the canonical basis of the space of matrices."""
-        return self._mat_space.basis
+class SquareMatricesLieAlgebraStruct(MatrixLieAlgebra):
+    """Lie algebra structure for square matrices."""
+
+    def __init__(self, space):
+        super().__init__(space=space, representation_dim=space.n)
 
     def basis_representation(self, matrix_representation):
```

Comment:
basis_representation and matrix_representation should be methods of the space, not methods of the LieBracket structure. Because they depend on a notion of basis, and the notion basis is a vector space property. We should be careful about which methods go in the Structure classes and which methods stay in the Spaces, as we did for the distinction between Metrics and Spaces.

Reply:
I fully agree here. This probably means we need to create an intermediate manifold structure inheriting from `VectorSpace` called e.g. `MatrixVectorSpace`, as this methods only make sense for the matrix case. Do you agree? (If yes, I'll do it in a different PR.)